### DISCUSSION TITLE
How edit/make animation

### DISCUSSION THREAD

**Discussion by realroot2185** [FyroxEngine/Fyrox#672] [Q&A]
For example I imported in blender `Idle.fbx` from the rpg tutorial.

How can edit it to make new animations?
I can move the points but when I play the animation they do not change (in blender).
Also they move without keeping the ratio with the model.

**Comment by mrDIMAS** [comment]
You should watch tutorials for Blender, it looks unrelated to the engine.

**Reply by realroot2185** [reply]
I see what type of files can I use in Fyrox? E.g. can I use blender files directly?
I am not experienced here so sorry if I ask something dumb, are animation universal or are there more types?

What was used to make rpg/fps tutorial animations?

**Reply by mrDIMAS** [reply]
You can't use blender files directly, only FBX and GLTF is supported (the latter requires `gltf` feature to be enabled).

**Reply by realroot2185** [reply]
Is it possible to export from blender in those formats and use them in Fyrox directly?

**Reply by mrDIMAS** [reply]
Yes

**Reply by realroot2185** [reply]
If I import in blender a file from the rpg tutorial e.g. Idle.fbx it looks different than an animation e.g. Hook.fbx downloaded from mixamo with the paladin and `Without Skin`.

If I use Hook.fbx in Fyrox it's all messed up.
Is there some step that am I missing?

Fyrox
<img width="424" height="529" alt="fyrox-file" src="https://github.com/user-attachments/assets/28244b3b-4c13-41ac-8b1f-16b0472fa638" />
Mixamo
<img width="391" height="515" alt="mixamo-file" src="https://github.com/user-attachments/assets/c0b1c537-ce15-47a2-bcf3-5cdf279a7be7" />

**Reply by realroot2185** [reply]
If use the paladin from mixamo site animations work fine.

I imported one fbx mixamo animation in blender and I exported it as fbx, only options for animations right?
To see if I can edit them.
When I try to preview it in fyrox it crashes:

```
[WARNING]: Unable to load options file ./data/models/paladin/test2.fbx.options for ./data/models/paladin/test2.fbx resource, fallback to defaults! Reason: Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
[INFO]: Trying to load "./data/models/paladin/test2.fbx"
[INFO]: FBX "./data/models/paladin/test2.fbx" loaded in 12 ms
	- Parsing - 7 ms
	- DOM Prepare - 2 ms
	- Conversion - 2 ms
[INFO]: Resource ./data/models/paladin/test2.fbx was loaded successfully!
[ERROR]: Failed to retarget animation External("./data/models/paladin/test2.fbx") for node Armature
[ERROR]: Failed to retarget animation External("./data/models/paladin/test2.fbx") for node Armature
[ERROR]: Failed to retarget animation External("./data/models/paladin/test2.fbx") for node Armature
```
Is this an issue?

**Reply by realroot2185** [reply]
If I use fyrox's `Idle.fbx` with mixamo model is all messed up.

**Reply by mrDIMAS** [reply]
There could be different coordinate system issue, re-export the animation from mixamo and use it instead.

**Reply by realroot2185** [reply]
How edit mixamo animations in blender in a way that I can use them in fyrox?

As I stated fyrox crashes, should I open an issue?

**Reply by realroot2185** [reply]
![hook-imported-blender](https://github.com/user-attachments/assets/52285c80-277f-4992-8b2b-5a38b069278e)
That is the Hook.fbx file imported and exported in bleder.
When using fyrox you will likely need to edit animations.
Could you maybe provide a way to fix that or something?
To export from blender to fyrox without problems.

**Reply by realroot2185** [reply]
If animations can be only fbx it seems a problem for libre games.
Will fyrox support also gltf?

**Reply by mrDIMAS** [reply]
At first, do not mix animations and meshes from examples and mixamo. Use one source only, mixamo for instance. Second, GLTF is already supported.

**Reply by realroot2185** [reply]
I am trying with only mixamo now as I stated, import/export in blender ruins them.

When I click `Animation Editor -> Import Animation` and then I select the model the menu does not show my `animation.glb` exported from blender.
Does it need some other suffix or something?

```
file animation.glb
animation.glb glTF binary model, version 2, length 317868 bytes
```
